src/types/sql.ts

```diff
@@ -0,0 +1,64 @@
+import { DataQuery } from '@grafana/schema';
+import { BuilderMode, QueryType, QueryBuilderOptions } from './queryBuilder';
+
+/**
+ * EditorType determines the query editor type.
+ */
+export enum EditorType {
+  SQL = 'sql',
+  Builder = 'builder',
+}
+
+export interface CHQueryBase extends DataQuery {
+  editorType: EditorType;
+  rawSql: string;
+
+  /**
+   * REQUIRED by backend for auto selecting preferredVisualisationType.
+   * Only used in explore view.
+   * src: https://github.com/grafana/sqlds/blob/main/query.go#L36
```

Comment:
nit: let's link to a specific commit, links to a branch are fragile since branches are mutable.